sources/zafira-web/src/main/java/com/qaprosoft/zafira/web/FilterController.java

```diff
@@ -47,63 +40,45 @@ public FilterController(FilterService filterService, Mapper mapper) {
-    public FilterType createFilter(@RequestBody @Valid FilterType filterType) {
-        filterType.setUserId(getPrincipalId());
-        filterType.getSubject().sortCriterias();
-
-        Filter filter = mapper.map(filterType, Filter.class);
-        if (filterService.isFilterExists(filter)) {
-            throw new IllegalOperationException(FILTER_CAN_NOT_BE_CREATED, ERR_MSG_FILTER_WITH_SUCH_NAME_ALREADY_EXISTS);
-        }
-        if (filter.isPublicAccess() && !isAdmin()) {
-            filter.setPublicAccess(false);
-        }
-
-        return mapper.map(filterService.createFilter(filter), FilterType.class);
+    public FilterDTO createFilter(@RequestBody @Valid FilterDTO filterDTO) {
+        filterDTO.getSubject().sortCriterias();
+        return mapper.map(filterService.createFilter(mapper.map(filterDTO, Filter.class), getPrincipalId(), isAdmin()), FilterDTO.class);
     }
 
     @ApiResponseStatuses
     @ApiOperation(value = "Get all public filters", nickname = "getAllPublicFilters", httpMethod = "GET", response = List.class)
-    @ApiImplicitParams({ @ApiImplicitParam(name = "Authorization", paramType = "header") })
+    @ApiImplicitParams({@ApiImplicitParam(name = "Authorization", paramType = "header")})
     @GetMapping("/all/public")
-    public List<FilterType> getAllPublicFilters() {
+    public List<FilterDTO> getAllPublicFilters() {
         List<Filter> publicFilters = filterService.getAllPublicFilters(getPrincipalId());
         return publicFilters.stream()
-                            .map(filter -> mapper.map(filter, FilterType.class))
+                            .map(filter -> mapper.map(filter, FilterDTO.class))
                             .collect(Collectors.toList());
     }
 
     @ApiResponseStatuses
-    @ApiOperation(value = "Update filter", nickname = "updateFilter", httpMethod = "PUT", response = FilterType.class)
-    @ApiImplicitParams({ @ApiImplicitParam(name = "Authorization", paramType = "header") })
+    @ApiOperation(value = "Update filter", nickname = "updateFilter", httpMethod = "PUT", response = FilterDTO.class)
+    @ApiImplicitParams({@ApiImplicitParam(name = "Authorization", paramType = "header")})
     @PutMapping()
-    public FilterType updateFilter(@RequestBody @Valid FilterType filterType) {
-        Filter filter = filterService.getFilterById(filterType.getId());
-        if (filter != null && !filter.getUserId().equals(getPrincipalId())) {
-            throw new IllegalOperationException(ILLEGAL_FILTER_ACCESS, ERR_MSG_ILLEGAL_FILTER_MODIFICATION);
-        }
-        filterType.getSubject().sortCriterias();
-        return mapper.map(filterService.updateFilter(mapper.map(filterType, Filter.class), isAdmin()), FilterType.class);
+    public FilterDTO updateFilter(@RequestBody @Valid FilterDTO filterDTO) {
+        filterDTO.getSubject().sortCriterias();
+        return mapper.map(filterService.updateFilter(mapper.map(filterDTO, Filter.class), getPrincipalId(), isAdmin()), FilterDTO.class);
```

Comment:
Same here (i'll skip this comment in next controllers, but please apply it everywhere)